packages/@sanity/components/src/selects/StyleSelect.js

```diff
@@ -176,39 +99,47 @@ class StyleSelect extends React.Component {
               {value && value.length == 1 && value[0].title}
               {!value && placeholder}
             </span>
-            <div className={styles.icon}>
+            <div className={styles.arrow}>
               <FaAngleDown color="inherit" />
             </div>
           </div>
         </div>
-
-        <div className={`${showList ? styles.listContainer : styles.listContainerHidden}`}>
-          {
-            items.map(item => {
-              const isSemiSelected = value && value.length > 1 && includes(value, item)
-              const isSelected = value && value.length === 1 && value[0] == item
-              const classNames = `
-                ${isSelected ? styles.itemSelected : styles.item}
-                ${isSemiSelected ? styles.itemSemiSelected : ''}
-              `
-              return (
-                <a className={classNames} key={item.key} title={item.title} onClick={() => this.handleSelect(item)}>
-                  <div className={styles.itemIcon}>
-                    {
-                      isSelected && <CircleCheckIcon />
-                    }
-                    {
-                      isSemiSelected && <CircleThinIcon />
-                    }
-                  </div>
-                  <div className={styles.itemContent}>
-                    {this.props.renderItem(item)}
-                  </div>
-                </a>
-              )
-            })
-          }
-        </div>
+        <Escapable onEscape={this.handleCloseList} />
+        {
+          showList && (
+            <List className={styles.list}>
+              {items.map((item, index) => {
+                const isSemiSelected = value && value.length > 1 && includes(value, item)
+                const isSelected = value && value.length === 1 && value[0] == item
+                const classNames = `
+                  ${isSelected ? styles.itemSelected : styles.item}
+                  ${isSemiSelected ? styles.itemSemiSelected : ''}
+                `
+                return (
+                  <Item
+                    key={item.key}
+                    title={item.title}
+                    data-index={index}
+                    onClick={this.handleSelect}
+                    className={classNames}
+                  >
+                    <div className={styles.itemIcon}>
+                      {
+                        isSelected && <CircleCheckIcon />
+                      }
+                      {
+                        isSemiSelected && <CircleThinIcon />
+                      }
+                    </div>
+                    <div className={styles.itemContent}>
+                      {this.props.renderItem(item)}
```

Comment:
Could you destruct `renderItem` from `this.props` like we do with the other props in the line in this method?